nodejs/webapi/routes/openapiservice.js

```diff
@@ -0,0 +1,76 @@
+
+const swaggerJsdoc = require("swagger-jsdoc");
+const swaggerUi = require("swagger-ui-express");
+
+// Function to determine if the port value is a named pipe or a port number
+function isNamedPipe(port) {
+  return isNaN(port);
+}
+
+// Function to format the base URL based on the port or named pipe
+function formatBaseUrl(port) {
+  const host = 'localhost';
+  if (isNamedPipe(port)) {
+    // Format for named pipe
+    return `http://${host}/${port}`;
+  } else {
+    // Format for port number
+    return `http://${host}:${port}`;
+  }
+}
+
+const portOrPipe = process.env.PORT || 3030; // Default to 3030 if not set
+const baseUrl = formatBaseUrl(portOrPipe);
+
+// Basic swagger definition
+const basicSwaggerDefinition = {
+  definition: {
+    openapi: "3.0.1",
+    info: {
+      title: "webapi",
+      description: `<a href="${baseUrl}/api-docs/swagger.json" target="_blank">${baseUrl}/api-docs/swagger.json</a>`,
+      version: "1.0",
+    },
+  },
+  apis: ["./app.js"],
+};
+
+// Improved swagger definition
+const improvedSwaggerDefinition = {
+  definition: {
+    openapi: "3.0.1",
+    info: {
+      title: "Weather Forecast API",
+      description: `<a href="${baseUrl}/api-docs/swagger.json" target="_blank">${baseUrl}/api-docs/swagger.json</a>`, 
```

Comment:
We should also include `An API that predicts upcoming weather` in the description.

Reply:
Done